### DISCUSSION TITLE
iterate send_chat_action

### DISCUSSION THREAD

**Discussion by PietroG72** [python-telegram-bot/python-telegram-bot#3598] [Q&A] [answered]
when my bot receives a message it take long time to respond because running a long computational task. In the meanwhile I send a "send_chat_action with typing", but after 5 seconds it stops. Is there a way to iterate send_chat_action, until a result is returned from computational task? I'm trying using asyncio.run and and similar but all my attempts are wrongs.

**Comment by Bibo-Joshi** [comment, marked as answer]
I did something like this [here](https://github.com/Bibo-Joshi/ptbstats/blob/eb84e37cd2f2de8524df6942a5dcac31e73183ba/ptbstats/_simplestats.py#L95-L106). Maybe that helps.
